doc/build-osx.md

```diff
@@ -179,6 +179,17 @@ brew install python
 You can [deploy](#3-deploy-optional) a `.zip` containing the Bitcoin Core application.
 It is required that you have `python` installed.
 
+---
+
+#### Relinking brew dependencies
+
+#### ***Try relinking brew installed dependencies - if they are not detected when building.***
+##### Example:
+
+```bash
+brew unlink qrencode && brew link qrencode
```

Comment:
Is this related to https://github.com/bitcoin/bitcoin/pull/30454/files#r1709145204?
> [l0rinc](https://github.com/l0rinc) [on Aug 8](https://github.com/bitcoin/bitcoin/pull/30454/files#r1709145204)
👍 GUI is working with cmake, but I had to execute a `brew link qt5 --force` as well after `brew install qt@5` (applies to Autotools as well)

Reply:
This PR isn't in response to and issue or comment. I just noticed it may be useful to have a note about relinking.

Reply:
> This PR isn't in response to any issue or comment. I just noticed it may be useful to have a note about relinking.

We don't really add general things to our build instructions that "may" be useful, especially if they aren't addressing a problem someone has encountered. In any case, in this instance, if your package manager is broken, or incorrectly installing dependencies such that they are broken, that isn't a Bitcoin Core issue, and I don't think we need to add anything here.

Reply:
In scenarios like this - the build just fails - a few extra lines for someone actually using the build instructions "may" find it useful. 

Additionally a similar note "may" be useful in a scenarios where linuxbrew is being used.

I dont see much downside in trying to make build instructions a little more friendly to users - with minimal experience - building from source.

I'd like to think it is encouraged.